services/advertise.js

```diff
@@ -207,8 +206,13 @@ module.exports = (args, cbk) => {
           probe.on('error', () => {});
 
           probe.on('probe_success', async ({route}) => {
+            //Exit when budget exceeded or within advertisement min amount
+            if(!!args.budget && (totalPaid >= args.budget || (args.budget - totalPaid) <= 10)) {
```

Comment:
maybe it should just be quitting when `(totalPaid + maxPossiblePaymentForNextAd > args.budget)` - if 10 is the keysend you also have to take into account they might pay routing fees on top of the 10 and that would cause more than the budget to be spent

Reply:
I made some changes, I think the only issue calculating up top is we are assuming the payment below will succeed down below?
No big deal?